sale_stock_prebook/models/sale.py

```diff
@@ -12,6 +12,12 @@ def _prepare_reserve_procurement_values(self, group_id=None):
         return values
 
     def _prepare_reserve_procurement(self, group):
+        no_prebook_routes = self.env["stock.location.route"].search(
+            [("no_prebook_stock", "=", True)]
+        )
+        for route in no_prebook_routes:
+            if route in self.product_id.route_ids:
+                return
```

Comment:
Please create some own methods. as example
```
# sale.order.line
def _should_prebook_stock(self):
    self.ensure_one()
    for route in self.product_id.route_ids:
        if route.no_sale_stock_prebook:
            return False
    return True

def _prepare_reserve_procurements(self, group):
    procurements = []
    for line in self:
        if not line._should_prebook_stock():
            continue
        procurements.append(line._prepare_reserve_procurement(group))
    return procurements
```

also change then the code in the sale.order to use
_prepare_reserve_procuements